dmipy/core/modeling_framework.py

```diff
@@ -1212,8 +1216,24 @@ def fit(self, acquisition_scheme, data,
             start = time()
             mt_fractions = np.empty(
                 np.r_[N_voxels, self.N_models], dtype=float)
+
+            S0_responses = []
+            for i, (m, s) in enumerate(zip(self.models,
+                                           self.S0_tissue_responses)):
+                if s is None:
+                    # If the S0 of the tissue is None, the model is composite
+                    # and the corresponding S0 is treated as the weighted
+                    # average of the S0 tissues modelled by the "submodels".
+                    s0 = 0.0
+                    for k, mm in enumerate(m.models):
```

Comment:
Nicely done, but while this may work here at the top level to the second level - just for this project - it means that the top level now has the logic to infer what is the S0 of a lower level distributed model.

In my opinion this is not so clean, as the we should be able to ask the distributed model what is it's S0 based on its signal volume fractions. This has the added benefit of it being to (potentially) generalize to more than 2 layers - we want to have the code as modular as possible.

Reply:
It makes sense.
If I understand the flow correctly, this could be done by moving the definition of `S0_responses` to the `__init__` method, where `self.S0_tissue_responses` will be defined inferring the different S0s from the "submodels".
Am I right?

Reply:
What I mean is that we should be able to say `S0 = m.S0_response(volume_fractions)` to any submodel, instead of having to manually sum up the s0 values in line 1229.

Hold on, I how is line 1229 even getting the `'partial_volume'` values? `mm` is just a DistributedModel instance right? not a `dict`? I don't thing this works like this, or am I wrong?